src/screens/RatingsModal/Init.tsx

```diff
@@ -0,0 +1,124 @@
+import React, { useCallback } from "react";
+import { TouchableOpacity } from "react-native";
+import { useDispatch, useSelector } from "react-redux";
+import { Trans } from "react-i18next";
+import { add } from "date-fns";
+import styled from "styled-components/native";
+import { Flex, Text, Button } from "@ledgerhq/native-ui";
+import useFeature from "@ledgerhq/live-common/lib/featureFlags/useFeature";
+import { setRatingsDataOfUserInStorage } from "../../logic/ratings";
+import {
+  ratingsDataOfUserSelector,
+  ratingsHappyMomentSelector,
+} from "../../reducers/ratings";
+import { setRatingsDataOfUser } from "../../actions/ratings";
+import { track } from "../../analytics";
+
+const NotNowButton = styled(TouchableOpacity)`
+  align-items: center;
+  justify-content: center;
+  margin-top: 24px;
+`;
+
+type Props = {
+  closeModal: Function;
+  setStep: Function;
+};
+
+const Init = ({ closeModal, setStep }: Props) => {
+  const ratingsFeature = useFeature("ratings");
+  const dispatch = useDispatch();
+  const ratingsDataOfUser = useSelector(ratingsDataOfUserSelector);
+  const ratingsHappyMoment = useSelector(ratingsHappyMomentSelector);
+  const goToEnjoy = useCallback(() => {
+    setStep("enjoy");
+    track("Satisfied", { source: ratingsHappyMoment.route_name });
+  }, [ratingsHappyMoment.route_name, setStep]);
+  const goToDisappointed = useCallback(() => {
+    setStep("disappointed");
+    track("Disappointed", { source: ratingsHappyMoment.route_name });
```

Comment:
I feel like this **logic** should be centralized in `useRatings()`log), so rather than having `useRatings()` returning `[isRatingsModalOpen, setRatingsModalOpenCallback]` it would return an object and a bunch of functions that are supposed to handle the logic (for instance here it would be something like `handleDisappointed`). That way the entire logic of the ratings is in one place rather than being widespread in components which in my opinion should only have visibility on the UI state and pseudo navigation (setStep).

That will also allow you to avoid duplicating code and have everything nicely factored (for instance the repetitive calls to
```
dispatch(setRatingsDataOfUser(ratingsDataOfUserUpdated));
setRatingsDataOfUserInStorage(ratingsDataOfUserUpdated);
```

Reply:
well actually then you would have to remove the `useEffect()` (that handles navigation changes) from `useRatings()` (and remove all other side effects from it)

Reply:
Totally agree. Thanks a lot for the feedback !